### DISCUSSION TITLE
Compile problems

### DISCUSSION THREAD

**Discussion by lampvideo** [WinMerge/winmerge#1159] [Q&A] [answered]
Hi all. I am using MSVC 2019 x86 and try to compile project by instuction but i see some unresolved links to _WINRT_GetRestrictedErrorInfo, _WINRT_RoGetActivationFactory, _WINRT_WindowsGetStringRawBuffer etc in ClipBoardHistory.obj. How can i fix it?

**Comment by sdottaka** [comment, marked as answer]
Recently, WinMerge has been using C++/WinRT to reference Clipboard history.
In the latest C++/WinRT, functions such as WINRT_RoGetActivationFactory are loaded dynamically at runtime, but in older versions, you need to link windowsapps.lib at build time.

You are probably using a slightly older Windows SDK.

In order to avoid linking errors in Visual Studio 2019, a fix was made in commit 1efd5ec to not use C++/WinRT if the version of Visual Studio used for the build is less than 2022.

**Comment by lampvideo** [comment]
Thank you! Everything is working now!
